RealmSwift/Tests/ObjectTests.swift

```diff
@@ -715,6 +850,220 @@ class ObjectTests: TestCase {
+        var ex = expectation(description: "no change notification")
+        ex.isInverted = true
+        var token = company.observe(keyPaths: ["employees"], { _ in
+            ex.fulfill()
+        })
+        try! realm.write({
+            employee.hired = true
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+
+        // Went over case with Dominic
+        // Expect a notification for "employees.hired" when "employee.hired" is changed
+        token = company.observe(keyPaths: ["employees.hired"], expectChange("employees", Int?.none, Int?.none))
+        try! realm.write({
+            XCTAssertTrue(employee.hired)
+            employee.hired = false
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+
+        // Expect no notification for "employees.hired" when "employee.age" is changed.
+        ex = expectation(description: "no change notification")
+        ex.isInverted = true
+        token = company.observe(keyPaths: ["employees.hired"], { _ in
+            ex.fulfill()
+        })
+        try! realm.write({
+            employee.age = 35
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+
+        // Expect notification for "employees.hired" when an employee is deleted.
+        token = company.observe(keyPaths: ["employees.hired"], expectChange("employees", Int?.none, Int?.none))
+        try! realm.write({
+            realm.delete(employee)
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+
+        // Expect notification for "employees" when an employee is added.
+        token = company.observe(keyPaths: ["employees"], expectChange("employees", Int?.none, Int?.none))
+        try! realm.write({
+            let employee2 = realm.create(SwiftEmployeeObject.self)
+            company.employees.append(employee2)
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+
+        // Expect notification for "employees" when an employee is reassigned.
+        token = company.observe(keyPaths: ["employees"], expectChange("employees", Int?.none, Int?.none))
+        try! realm.write({
+            let employee3 = realm.create(SwiftEmployeeObject.self)
+            company.employees[0] = employee3
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+
+        // Expect no notification for "employees" when "company.name" is changed
+        ex = expectation(description: "no change notification")
+        ex.isInverted = true
+        token = company.observe(keyPaths: ["employees"], { _ in
+            ex.fulfill()
+        })
+        try! realm.write({
+            company.name = "changed"
+        })
+        waitForExpectations(timeout: 0.1, handler: nil)
+        token.invalidate()
+    }
+
+    func testLinkPropertyKeyPathNotifications() {
```

Comment:
Same as above. Might be nice to split this and name each expectation / test with regards to what is tested.